src/datasets/table.py

```diff
@@ -0,0 +1,591 @@
+import copy
+from functools import wraps
+from typing import List, TypeVar, Union
+
+import pyarrow as pa
+
+
+def wraps_arrow_table_method(arrow_table_method):
```

Comment:
```suggestion
def inject_arrow_table_documentation(arrow_table_method):
```

Reply:
much clearer, thanks
doing this change now